test/dynamo/test_generator.py

```diff
@@ -0,0 +1,672 @@
+        gen = fn(t)
+        self.assertEqual(list(gen), [t + 2, t + 3])
+
+    def test_dynamo_disable_generator(self):
+        @torch._dynamo.disable
+        def main_gen(t):
+            yield t + 1
+            yield t + 2
+            yield t + 3
+
+        @torch.compile(backend="eager", fullgraph=False)
+        def fn(t):
+            gen = main_gen(t)
+            return list(gen)
+
+        t = torch.randn(2)
+        y = fn(t)
+        self.assertEqual(y, [t + 1, t + 2, t + 3])
+
+    def test_dynamo_disable_sub_generator(self):
+        @torch._dynamo.disable
+        def subgen(t):
+            yield t + 2
+            yield t + 3
+
+        def main_gen(t):
+            yield t + 1
+            yield from subgen(t)
+
+        @torch.compile(backend="eager", fullgraph=False)
+        def fn(t):
+            gen = main_gen(t)
+            return list(gen)
+
+        t = torch.randn(2)
+        y = fn(t)
+        self.assertEqual(y, [t + 1, t + 2, t + 3])
+
+    def test_graph_break_outside_generator(self):
+        def whoo(t):
+            yield t + 1
+            yield t + 2
+
+        @torch.compile(backend="eager", fullgraph=False)
+        def fn(t):
+            gen = whoo(t)
+            x = next(gen)
+            torch._dynamo.graph_break()
+            y = next(gen)
+            return x + y
+
+        t = torch.randn(2)
+        y = fn(t)
+        self.assertEqual(y, (t + 1) + (t + 2))
+
+    def test_graph_break_before_calling_generator(self):
+        def whoo(t):
+            for perm in itertools.product(itertools.permutations((0, 1, 2)), repeat=1):
+                yield sum(perm[0])
+
+        def fn(t):
+            s = 0
+            for b, p in itertools.product(whoo(t), itertools.permutations((4, 5))):
+                s += b
+            return s
+
+        t = torch.randn(2)
+        expected = fn(t)
+        got = torch.compile(backend="eager", fullgraph=False)(fn)(t)
+        self.assertEqual(expected, got)
+
+    @unittest.expectedFailure
+    def test_generator_with_side_effects(self):
+        i = 0
+
+        def whoo(t):
+            nonlocal i
+            for j in range(5):
+                i += 1
+                yield t + j
+
+        def fn(t):
+            return whoo(t), t.sin()
+
+        t = torch.randn(2)
+        with self.assertRaises(Unsupported):
+            fn(t)
+
+    @unittest.expectedFailure
+    def test_subgenerator_with_side_effects(self):
+        i = 0
+
+        def subgen(t):
+            nonlocal i
+            i += 1
+            yield t
+            i += 1
+            yield t + 1
+
+        def whoo(t):
+            nonlocal i
+            yield from subgen(t)
+            i += 1
+            yield t + 2
+            i += 1
+            yield t + 3
+            i += 1
+            yield t + 4
+
+        def fn(t):
+            return whoo(t), t.sin()
+
+        with self.assertRaises(Unsupported):
+            self._compile_check(fn)
+
+    @unittest.expectedFailure
```

Comment:
I see a bunch of unittest.expectedFailure. I assume we're going to come back later and remove them once we support returning generators from a Dynamo region?

Reply:
The expected failures are either from CPython test cases, which uses unittest or removed on follow-up PRs in the stack.